documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -3260,3 +3271,192 @@ resourceDefinition": {
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : The response code {0} is defined without a x-ms-error-response.
+
+**Description** :  If defines response code 4xx or 5xx ,  x-ms-error-response:true is required. There is one exception: a HEAD operation with 404 SHOULD have x-ms-error-response:false.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: As some SDK may treat the 4xx or 5xx as exceptional code, if don't specified x-ms-error-response:true, the SDK will not handle the error schema correctly instead it will throw an exception.
+
+**How to fix the violation**: Add the x-ms-error-response:true for the error response code.
+
+The following would be valid:
+
+```json
+ "responses": {
+    "404": {
+      "description": "Not Found",
+      "x-ms-error-response": true
+    }
+ }
+```
+The following would be invalid:
+
+```json
+ "responses": {
+    "404": {
+      "description": "Not Found"
+    }
+ }
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4033"></a>R4033 UniqueModelName
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The model name {0} is duplicated with {1} .
+
+**Description** :  Do not use case sensitivity to differentiate model names.
+
+**CreatedAt**: February 23, 2021
+
+**LastModifiedAt**: February 23, 2021
+
+**Why this rule is important**: In Python SDK, model names are converted to forms starting with capital. So all of "AAAA", "aaaa", "Aaaa" will be transformed to "Aaaa". So differentiating model names by their case sensitivities would break Python SDK generation.
+
+**How to fix the violation**: Rename the duplicate name .
+
+The following would be invalid:
+
+```json
+"definitions": {
+  "SKU": {
+    "type": "string",
+    "description": "SKU in request"
+  },
+  "sku": {
+    "type": "string",
+    "description": "SKU in response"
+  }
+}
+```
+
+The following would be valid:
+
+```json
+"definitions": {
+  "requestSKU": {
+    "type": "string",
+    "description": "SKU in request"
+  },
+  "responseSKU": {
+    "type": "string",
+    "description": "SKU in response"
+  }
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4034"></a>R4034 AzureResourceTagsSchemaValidation
+
+**Category** : SDK Error
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** :  The resource "{0}" property tags schema is different with the common type definition.
```

Comment:
The property tags in the resource "{0}" does not follow/adopt the common type definition.